src/metabase/api/user.clj

```diff
@@ -160,9 +160,30 @@
   [clauses]
   (dissoc clauses :order-by :limit :offset))
 
+(defn- group-ids-for-manager
+  "Given a `user-id` return a list of group-ids of which the user is a group manager."
+  [user-id]
+  (map :group_id
+       (t2/query
+        {:select-distinct [:permissions_group_membership.group_id]
+         :from  [:permissions_group_membership]
+         :where [:and [:= :permissions_group_membership.user_id user-id]
+                 [:= :permissions_group_membership.is_group_manager true]
+                 [:not= :permissions_group_membership.group_id (:id (perms-group/all-users))]]})))
+
+(defn- check-manager-of-group
+  "Check that the `*current-user*` is a manager of the group with id `group-id`, or throw a 403."
+  [group-id]
+  {:pre [(integer? group-id)]}
+  (api/check-403
+   (or
+    api/*is-superuser?*
+    ((set (group-ids-for-manager api/*current-user-id*)) group-id))))
```

Comment:
For checking if a user is a manager of a specific group I don't think it's necessary to read their entire group list. You could just have a query that checks if a row exists for the user ID and group ID where `is_group_manager` is `true`